File: app/src/main/java/in/hbb20/countrycodepickerproject/DefaultCountryFragment.java
package in.hbb20.countrycodepickerproject;


import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.hbb20.CountryCodePicker;


/**
 * A simple {@link Fragment} subclass.
 */
public class DefaultCountryFragment extends Fragment {


    EditText editTextDefaultPhoneCode, editTextDefaultNameCode;
    Button buttonSetNewDefaultPhoneCode, buttonSetNewDefaultNameCode, buttonResetToDefault;
    CountryCodePicker ccp;
    Button buttonNext;

    public DefaultCountryFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_default_country, container, false);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        assignViews();
        editTextWatcher();
        addClickListeners();
    }

    private void addClickListeners() {
        buttonSetNewDefaultPhoneCode.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int code = -1;
                try {
                    code = Integer.parseInt(editTextDefaultPhoneCode.getText().toString());
                    ccp.setDefaultCountryUsingPhoneCode(code);
                    Toast.makeText(getActivity(), "Now default country is " + ccp.getDefaultCountryName() + " with phone code " + ccp.getDefaultCountryCode(), Toast.LENGTH_LONG).show();
                } catch (Exception ex) {
                    Toast.makeText(getActivity(), "Invalid number format", Toast.LENGTH_LONG).show();
                }
            }
        });

        buttonSetNewDefaultNameCode.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String nameCode;
                try {
                    nameCode = editTextDefaultNameCode.getText().toString();
                    ccp.setDefaultCountryUsingNameCode(nameCode);
                    Toast.makeText(getActivity(), "Now default country is " + ccp.getDefaultCountryName() + " with phone code " + ccp.getDefaultCountryCode(), Toast.LENGTH_LONG).show();
                } catch (Exception ex) {

                }
            }
        });

        buttonResetToDefault.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ccp.resetToDefaultCountry();
            }
        });

        buttonNext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ((ExampleActivity) getActivity()).viewPager.setCurrentItem(((ExampleActivity) getActivity()).viewPager.getCurrentItem() + 1);
            }
        });
    }

    private void editTextWatcher() {
        editTextDefaultPhoneCode.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                buttonSetNewDefaultPhoneCode.setText("set " + s + " as Default Country Code");
            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });

        editTextDefaultNameCode.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                buttonSetNewDefaultNameCode.setText("set '" + s + "' as Default Country Name Code");
            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });
    }

    private void assignViews() {
        editTextDefaultPhoneCode = (EditText) getView().findViewById(R.id.editText_defaultCode);
        ccp = (CountryCodePicker) getView().findViewById(R.id.ccp);
        buttonSetNewDefaultPhoneCode = (Button) getView().findViewById(R.id.button_setDefaultCode);
        buttonResetToDefault = (Button) getView().findViewById(R.id.button_resetToDefault);

        editTextDefaultNameCode = (EditText) getView().findViewById(R.id.editText_defaultNameCode);
        buttonSetNewDefaultNameCode = (Button) getView().findViewById(R.id.button_setDefaultNameCode);

        buttonNext = (Button) getView().findViewById(R.id.button_next);
    }
}

File: app/src/main/java/in/hbb20/countrycodepickerproject/DefaultCountryFragment.kt
package `in`.hbb20.countrycodepickerproject

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.hbb20.CountryCodePicker

/**
 * A simple [Fragment] subclass.
 */
class DefaultCountryFragment: Fragment() {

	private lateinit var editTextDefaultPhoneCode: EditText
	private lateinit var editTextDefaultNameCode: EditText
	private lateinit var buttonSetNewDefaultPhoneCode: Button
	private lateinit var buttonSetNewDefaultNameCode: Button
	private lateinit var buttonResetToDefault: Button
	private lateinit var ccp: CountryCodePicker
	private lateinit var buttonNext: Button

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
		// Inflate the layout for this fragment
		return inflater.inflate(R.layout.fragment_default_country, container, false)
	}

	override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
		super.onViewCreated(view, savedInstanceState)

		assignViews()
		editTextWatcher()
		addClickListeners()
	}

	private fun addClickListeners() {
		buttonSetNewDefaultPhoneCode.setOnClickListener {
			try {
				val code = Integer.parseInt(editTextDefaultPhoneCode.text.toString())
				@Suppress("DEPRECATION")
				ccp.setDefaultCountryUsingPhoneCode(code)
				Toast.makeText(activity, "Now default country is " + ccp.defaultCountryName + " with phone code " + ccp.defaultCountryCode, Toast.LENGTH_LONG).show()
			} catch (ex: Exception) {
				Toast.makeText(activity, "Invalid number format", Toast.LENGTH_LONG).show()
			}
		}

		buttonSetNewDefaultNameCode.setOnClickListener {
			val nameCode: String
			try {
				nameCode = editTextDefaultNameCode.text.toString()
				ccp.setDefaultCountryUsingNameCode(nameCode)
				Toast.makeText(activity, "Now default country is " + ccp.defaultCountryName + " with phone code " + ccp.defaultCountryCode, Toast.LENGTH_LONG).show()
			} catch (ex: Exception) {

			}
		}

		buttonResetToDefault.setOnClickListener { ccp.resetToDefaultCountry() }

		buttonNext.setOnClickListener { (activity as ExampleActivity).viewPager.currentItem = (activity as ExampleActivity).viewPager.currentItem + 1 }
	}

	private fun editTextWatcher() {
		editTextDefaultPhoneCode.addTextChangedListener(object: TextWatcher {
			override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
			}

			override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
				buttonSetNewDefaultPhoneCode.text = "set $s as Default Country Code"
			}

			override fun afterTextChanged(s: Editable) {
			}
		})

		editTextDefaultNameCode.addTextChangedListener(object: TextWatcher {
			override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
			}

			override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
				buttonSetNewDefaultNameCode.text = "set '$s' as Default Country Name Code"
			}

			override fun afterTextChanged(s: Editable) {
			}
		})
	}

	private fun assignViews() {
		editTextDefaultPhoneCode = view!!.findViewById(R.id.editText_defaultCode)
		ccp = view!!.findViewById(R.id.ccp)
		buttonSetNewDefaultPhoneCode = view!!.findViewById(R.id.button_setDefaultCode)
		buttonResetToDefault = view!!.findViewById(R.id.button_resetToDefault)

		editTextDefaultNameCode = view!!.findViewById(R.id.editText_defaultNameCode)
		buttonSetNewDefaultNameCode = view!!.findViewById(R.id.button_setDefaultNameCode)

		buttonNext = view!!.findViewById(R.id.button_next)
	}
}